readthedocs/embed/views.py

```diff
@@ -143,39 +143,48 @@ def get(self, request):
         version = self._get_version()
         doc = request.GET.get('doc')
         section = request.GET.get('section')
-        docpath = request.GET.get('path')
+        path = request.GET.get('path')
         url = request.GET.get('url')
 
         if self.unresolved_url:
-            project, _, _, docpath, section = self.unresolved_url
+            unresolved = self.unresolved_url
+            project = unresolved.project
+            path = unresolved.filename
+            section = unresolved.fragment
+
             # update docpath from unresolved URL
-            if docpath.endswith('/'):
+            if path.endswith('/'):
                 doc = doc.path.rstrip('/')
             else:
-                doc = docpath.split('.html', 1)[0]
+                doc = path.split('.html', 1)[0]
 
         # Check for None here becuase '' is valid for no filename
         if not project.slug and doc is not None:
-            return Response({
-                'error': (
-                    'Invalid Arguments. '
-                    'Please provide "project" and "doc", or "url" GET argument.'
-                )
-            },
+            return Response(
+                {
+                    'error': (
+                        'Invalid Arguments. '
+                        'Please provide "project" and "doc", or "url" GET argument.'
+                    )
+                },
                 status=status.HTTP_400_BAD_REQUEST
             )
 
         return do_embed(
-            project=project, version=version,
-            doc=doc, section=section, path=docpath, url=url
+            project=project,
+            version=version,
+            doc=doc,
+            section=section,
+            path=path,
+            url=url,
         )
 
 
 class EmbedAPI(SettingsOverrideObject):
     _default_class = EmbedAPIBase
 
 
-def do_embed(project, version, doc, section=None, path=None, url=None):
+def do_embed(*, project, version, doc=None, path=None, section=None, url=None):
```

Comment:
We are changing the signature of the function here. `doc` was _required_ previously and now it's not. Why it's not required anymore?

Reply:
doc is the pagename, that's sphinx specific (isn't required when the project is mkdocs type as you can see below), and isn't required to get the response if path and section are given.